src/main.rs

```diff
@@ -81,15 +82,25 @@ enum Commands {
 // Looking at https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout
 // it is a little complicated.
 // The current setup is to write all progress information to STDERR and all data to STDOUT.
-macro_rules! progress {
-    // Match a format string followed by any number of arguments
-    ($fmt:expr $(, $args:expr)*) => {{
-        // Use the `format!` macro to format the string with the arguments
-        let mut message = format!($fmt $(, $args)*);
-        // Print the formatted string to the console with a newline
-        message.push('\n');
-        tokio::io::stderr().write_all(message.as_ref()).await.unwrap();
-    }};
+struct OutWriter {
+    stderr: Mutex<tokio::io::Stderr>,
+}
```

Comment:
Ah, whoops.  Didn't read my email in time and I made a competing PR with significantly less complexity: https://github.com/n0-computer/iroh/pull/804